File: shopapp-angular/src/app/DTO/orders/orderDto.ts
import {
  IsString,
  IsNotEmpty,
  IsPhoneNumber,
  IsDate
} from 'class-validator';
import { Product } from 'src/app/models/product';

export class OrderDto {

  user_id: number

  fullname: string

  email: string

  phone_number: string

  address: string

  note: string

  total_price: number

  payment_method: string

  shipping_method: string

  shipping_address: string

  coupon_code: string

  cart_items: { product_id: number, quantity: number }[]

  constructor(data: any) {
    this.user_id = data.user_id;
    this.fullname = data.fullname;
    this.email = data.email;
    this.phone_number = data.phone_number;
    this.address = data.address;
    this.note = data.note;
    this.total_price = data.total_price;
    this.payment_method = data.payment_method;
    this.shipping_method = data.shipping_method;
    this.shipping_address = data.shipping_address;
    this.coupon_code = data.coupon_code;
    this.cart_items = data.cart_items || [];
  }
}

File: shopapp-server/src/main/java/com/project/shopapp/dto/OrderDTO.java
package com.project.shopapp.dto;


import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class OrderDTO {

    private int id;

    @JsonProperty("user_id")
    @NotNull
    private int userId;

    @JsonProperty("fullname")
    @NotBlank(message = "Full name is required")
    private String fullName;

    @NotBlank(message = "Email is required")
    private String email;

    @JsonProperty("phone_number")
    @NotBlank(message = "Phone number is required")
    private String phoneNumber;

    @NotBlank(message = "Address is required")
    private String address;

    private String note;

    private String status;

    @JsonProperty("total_price")
    @NotNull(message = "Total price is not null")
    private float totalPrice;

    @JsonProperty("shipping_method")
    @NotBlank(message = "Shipping method is required")
    private String shippingMethod;

    @JsonProperty("shipping_address")
    @NotBlank(message = "Shipping address is required")
    private String shippingAddress;

    @JsonProperty("shipping_date")
    private LocalDate shippingDate;

    @JsonProperty("payment_method")
    @NotBlank(message = "Payment method is required")
    private String paymentMethod;

    @JsonProperty("cart_items")
    private List<CartItemsDTO> cartItems;
}
